components/sup/src/manager/mod.rs

```diff
@@ -1741,6 +1779,23 @@ impl Manager {
     }
 }
 
+fn tls_config(tls_files: (PathBuf, PathBuf)) -> Result<ServerConfig> {
+    let (kp, cp) = tls_files;
+    let mut config = ServerConfig::new(NoClientAuth::new());
+    let key_file = &mut BufReader::new(File::open(&kp).unwrap());
+    let cert_file = &mut BufReader::new(File::open(&cp).unwrap());
+
+    let cert_chain = pemfile::certs(cert_file)
+        .map_err(|_| sup_error!(Error::InvalidCertFile(cp, String::new())))?;
+    let mut keys = pemfile::rsa_private_keys(key_file)
+        .map_err(|_| sup_error!(Error::InvalidKeyFile(kp, String::new())))?;
+    config
+        .set_single_cert(cert_chain, keys.remove(0))
```

Comment:
If keys doesn't have any items, this will panic.

As an alternative, instead of declaring `keys` to be `mut`, to use `pop` to grab the first key directly, e.g. something like:

```rust
let key = pemfile::rsa_private_keys(key_file)?
    .pop(0)
    .ok_or(<error for no keys>);
```
You can do this, because the keys vector isn't available anywhere else, so you can "mutate" it all you want; the only thing that matters is the immutable key that you get at the end.

That way you don't have any unnecessary mutability and you can handle the "empty keys".

Reply:
All of the validation for `keys` is done up front, before this function is ever called.  If keys doesn't contain at least item, we'll return an error before we ever make it here.

Reply:
The validation is actually incorporated into this function now, and is much cleaner.